src/components/edit/earthEngine/StyleSelect.js

```diff
@@ -11,40 +11,48 @@ import { setParams } from '../../../actions/layerEdit';
 import { EE_BUFFER } from '../../../constants/layers';
 import styles from '../styles/LayerDialog.module.css';
 
-const getStepsFromParams = ({ palette, min }) =>
-    palette.split(',').length - (min === 0 ? 1 : 2);
+const minSteps = 3;
+const maxSteps = 9;
 
-const isValidSteps = (newSteps, params) => {
-    const steps =
-        newSteps !== undefined ? newSteps : getStepsFromParams(params);
-    return steps > 0 && steps < 8; // Valid steps: 1-7
-};
+const isValidParams = ({ min, max }) =>
+    !Number.isNaN(min) && !Number.isNaN(max) && max > min;
 
 const StyleSelect = ({ unit, params, setParams }) => {
-    const [steps, setSteps] = useState();
     const { min, max, palette } = params;
+    const [steps, setSteps] = useState(palette.split(',').length);
+    const legend = isValidParams(params) && createLegend(params);
 
     const onStepsChange = useCallback(
-        newSteps => {
-            const steps = newSteps === '' ? '' : parseInt(newSteps, 10);
-
-            if (isValidSteps(steps, params)) {
-                const { min, max, palette } = params;
+        steps => {
+            if (steps >= minSteps && steps <= maxSteps) {
                 const scale = getColorScale(palette);
-                const classes =
-                    (steps == 1 && min == 0 ? 2 : steps) + (min == 0 ? 1 : 2);
-                const newPalette = getColorPalette(scale, classes);
+                const newPalette = getColorPalette(scale, steps);
 
                 if (newPalette) {
-                    setParams(min, max, newPalette);
+                    setParams({ palette: newPalette });
                 }
-
-                setSteps(steps);
             }
+
+            setSteps(steps);
         },
-        [params, setParams]
+        [palette, setParams]
     );
 
+    let help;
+
+    if (Number.isNaN(min)) {
+        help = i18n.t('Min value is required');
+    } else if (Number.isNaN(max)) {
+        help = i18n.t('Max value is required');
+    } else if (max <= min) {
+        help = i18n.t('Max should be higher than min');
```

Comment:
`higher` -> `larger`, `greater`, `bigger`? Another grammar question for the native speakers 🙂  @jenniferarnesen

Reply:
Since the unit depends on the layer (e.g. elevation uses meters and population uses people per hectare) I guess we should use the most generic phrase here unless we want to swap it out based on the layer (which is a bit overkill IMHO).

Reply:
Hmm, maybe greater?

Reply:
Joe replied: 

> generally speaking, I would say greater, that’s the “mathematical” comparison term. I would only use bigger or larger if the numbers referred to something physical (a surface area etc.). so easiest to go with greater

Reply:
Changed to "Max should be greater than min"